input_manager/src/EditorInputManager.cpp

```diff
@@ -0,0 +1,163 @@
+                auto event = utils::pop(arg);
+                BOOST_LOG_TRIVIAL(debug) << "Released button in " << event.xkey.time;
+            }
+        }
+    });
+    register_handler(InputEventType::key_press, [this](InputManagerImplementation::event_stack_type& arg) {
+        if (arg.empty()) {
+            BOOST_LOG_TRIVIAL(error) << "Event stack is empty";
+            return;
+        }
+
+        if (this->rendering_context_ == nullptr) {
+            BOOST_LOG_TRIVIAL(error) << "Rendering context is nullptr";
+            return;
+        }
+        auto event = utils::pop(arg);
+        BOOST_LOG_TRIVIAL(debug) << event.xkey.keycode;
+
+        auto key_sym = XLookupKeysym(&(event.xkey), 0);
+
+        switch(key_sym) {
+            case XK_w:
+            case XK_W:
+            {
+                rendering_context_->tvec = rendering_context_->tvec + rcbe::math::Vector3d {0.0, 0.0, 1.0};
+            }
+                break;
+            case XK_s:
+            case XK_S:
+            {
+                rendering_context_->tvec = rendering_context_->tvec + rcbe::math::Vector3d {0.0, 0.0, -1.0};
+            }
+                break;
+            case XK_a:
+            case XK_A:
+            {
+                rendering_context_->tvec = rendering_context_->tvec + rcbe::math::Vector3d {1.0, 0.0, 0.0};
+            }
+                break;
+            case XK_d:
+            case XK_D:
+            {
+                rendering_context_->tvec = rendering_context_->tvec + rcbe::math::Vector3d {-1.0, 0.0, 0.0};
+            }
+                break;
+            case XK_Up:
+            {
+                rendering_context_->rvec = rendering_context_->rvec + rcbe::math::Vector3d {-1.0, 0.0, 0.0};
+            }
+                break;
+            case XK_Down:
+            {
+                rendering_context_->rvec = rendering_context_->rvec + rcbe::math::Vector3d {1.0, 0.0, 0.0};
+            }
+                break;
+            case XK_Left:
+            {
+                rendering_context_->rvec = rendering_context_->rvec + rcbe::math::Vector3d {0.0, -1.0, 0.0};
+            }
+                break;
+            case XK_Right:
+            {
+                rendering_context_->rvec = rendering_context_->rvec + rcbe::math::Vector3d {0.0, 1.0, 0.0};
+            }
+                break;
+            case XK_Page_Up: {
+                rendering_context_->tvec = rendering_context_->tvec + rcbe::math::Vector3d {0.0, -1.0, 0.0};
+            }
+            break;
+            case XK_Page_Down: {
+                rendering_context_->tvec = rendering_context_->tvec + rcbe::math::Vector3d {0.0, 1.0, 0.0};
+            }
+                break;
+            default:
+                BOOST_LOG_TRIVIAL(info) << "Unhandled KeySym " << key_sym;
+                break;
+        }
+
+        arg.push(event);
+    });
+    register_handler(InputEventType::button_release, [this](InputManagerImplementation::event_stack_type& arg) {
```

Comment:
I have a suspicion that you should pop the event two times since that is a release event?

Reply:
Nope, you are wrong. It crashes if done as you suggest.